lib/netapi/NetAPI.cc

```diff
@@ -55,18 +55,43 @@ SObj network_socket_connect_tcp(Timeout *timeout,
 		Debug::log("Host capability does not authorise a TCP connection");
 		return nullptr;
 	}
+
+	bool isIPv6 = false;
 	firewall_permit_dns();
 	NetworkAddress address = network_host_resolve(host->hostname, UseIPv6);
 	firewall_permit_dns(false);
-	if (address.kind == NetworkAddress::AddressKindInvalid)
+	if (address.kind == NetworkAddress::AddressKindIPv6)
 	{
+		isIPv6 = true;
+	}
+	else if (address.kind != NetworkAddress::AddressKindIPv4)
+	{
+		// Do not check for `NetworkAddress::AddressKindInvalid`
```

Comment:
This is currently fragile. We need a compiler issue to track the fact that we’re dead-store eliminating initialisation of sret parameters for cross-compartment calls. Then the original code would be correct. The neq code is correct only if we happen to have specific stack contents at the start.